### ISSUE TITLE
storybook/no-uninstalled-addons false positives for submodule

### ISSUE THREAD

**Issue by guilhermetod** [storybookjs/eslint-plugin-storybook#148]
**Describe the bug**
Potentially unwanted error report when using a submodule from a dependency.

Ie, if package.json had these packages:

```json
{
    "@nx/react": "17.2.8",
    "@storybook/addon-essentials": "^7.5.3",
    "@storybook/addon-interactions": "^7.5.3",
}
```

We get a false positive as described below:

```ts
  addons: [
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
    '@nx/react/plugins/storybook', // This has an error
    '@nx/react', // This will not have an error, but it's incorrect path
  ],
```

Keep in mind that it's not possible to install `@nx/react/plugins/storybook`, so my suggestion if to check for partial dep match instead of full path.

**To Reproduce**
1. Start nx monorepo
2. Add react application
3. Set up storybook with nx generator
4. Configure eslint-plugin-storybook with mentioned rule
5. Check error on `main.ts`

**Expected behavior**

**Screenshots**
![image](https://github.com/storybookjs/eslint-plugin-storybook/assets/53096593/414c49b5-2afd-4db4-811e-16c8150eab50)
